gateway/api/serializers.py

```diff
@@ -138,9 +138,9 @@ def create(self, validated_data):
         pass
 
 
-class RunExistingJobSerializer(serializers.ModelSerializer):
+class RunAndRunExistingJobSerializer(serializers.ModelSerializer):
     """
-    Job serializer for the /run_existing end-point
+    Job serializer for the /run and /run_existing end-point
```

Comment:
long-term thought: would it make sense to refactor the API to just use a single `/run` endpoint and distinguish within the call whether the program/function/pattern/job/building_block/element/quantum_magic has already been uploaded or not?

Reply:
Definitely I think it's something that we can propose. I had in mind to continue the improvements in the gateway from an code organization point of view: https://github.com/Qiskit-Extensions/quantum-serverless/pull/1286#discussion_r1567511446

I considered too that we could refactor `/run_existing` end-point to `/upload` + `/run` in the client or something like that. Feel free to open an issue or a discussion because I think it deserves the times the refactorization.